### PR TITLE
PLANNER-2448 Restore the possibility to configure KieBase options

### PR DESCRIPTION
**Pull request by triceo**
@Christopher-Chianelli Can you please check that there is nothing here to do wrt. Quarkus? I only re-enable this functionality outside of Quarkus and I'm not sure if/how we support that with Quarkus.

### CODE REVIEW

File: optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/ScoreDirectorFactoryFactory.java [hunk 1]
```diff
@@ -309,7 +312,9 @@ private void validateNoDroolsAlphaNetworkCompilation() {
                         kieHelper.addResource(new FileSystemResource(scoreDrlFile));
                     }
                 }
-                kieBase = kieHelper.build(ExecutableModelProject.class, KieBaseMutabilityOption.DISABLED);
+                KieBaseConfiguration kieBaseConfiguration = buildKieBaseConfiguration(KieServices.get());
+                kieBaseConfiguration.setOption(KieBaseMutabilityOption.DISABLED); // Performance improvement.
+                kieBase = kieHelper.build(ExecutableModelProject.class, kieBaseConfiguration);
```
Comment (rsynek) [inline optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/ScoreDirectorFactoryFactory.java:317]: Cannot resolve method 'build(java.lang.Class<org.drools.modelcompiler.ExecutableModelProject>, org.kie.api.KieBaseConfiguration)'

Does this PR depend on some latest changes in Drools?

Comment (triceo) [reply optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/ScoreDirectorFactoryFactory.java:317]: Yes, merged into Kogito today. Will be available tomorrow.

**Review by rsynek — APPROVED** [review]
Good to merge as soon as the new Kogito artifacts are available.

**Comment by Christopher-Chianelli** [comment]
@triceo Quarkus does not go on that code path for DRL. If `drl-integration-test` passes, then everything probably good.

**Pull request merged**